source/solvers/heat_transfer_scratch_data.cc

```diff
@@ -59,16 +64,24 @@ HeatTransferScratchData<dim>::allocate()
     std::vector<std::vector<double>>(n_q_points, std::vector<double>(n_dofs));
 
   // Allocate memory for the physical properties
+  fields.insert(
+    std::pair<field, std::vector<double>>(field::temperature, n_q_points));
+  fields.insert(
+    std::pair<field, std::vector<double>>(field::previous_temperature,
+                                          n_q_points));
+  fields.insert(
+    std::pair<field, std::vector<double>>(field::shear_rate, n_q_points));
   specific_heat        = std::vector<double>(n_q_points);
   density              = std::vector<double>(n_q_points);
   thermal_conductivity = std::vector<double>(n_q_points);
+  viscosity            = std::vector<double>(n_q_points);
```

Comment:
What to these lines do that's different from lines 36 to 39? I think I'm a bit confused

Reply:
Awesome I'll make the changes for Navier-stokes :)